dts-plugin fails to download types archive from localhost on windows

### Describe the bug

Using a minimal host configuration to load TypeScript definitions fails on windows when the remote is on localhost:
```typescript
federation({
    name: 'host',
    remotes: {
        '@some-company/my-app': 'http://localhost:4173/mf-manifest.json'
    },
    dts: {
        generateTypes: false,
        consumeTypes: true
    }
})
```

This is logged to the `./.mf/typesGenerate.log` file:
`[ERROR] downloadTypesArchive - Error during types archive download: Network error: Unable to download federated mocks for '@some-company/my-app' from 'http://localhost:4173/@mf-types.zip' because 'connect ECONNREFUSED 127.0.0.1:4173'`

This error happens because the dts-plugin enforces IPv4 for the remote by default. This is an error prone default in my opinion. 
https://github.com/module-federation/core/blob/091e7e6f79379089d7e86f3fe88fdb65e11f78a4/packages/dts-plugin/src/core/lib/utils.ts#L134

In my case the remote is a locally installed Vite.js project. Vite.js binds the dev/preview server to localhost. (default behavior)
Windows resolves `localhost` to the IPv6 ::1 address. Linux/macOS resolves it to the IPv4 127.0.0.1 address. (usual default behavior) 

Is there any specific reason why we enforce the IP family in the first place?
I suggest introducing the `0` value for the `family` option and defaulting to it. 
DNS hostnames can then be resolved to both IPv4 and IPv6 addresses. 

The `family` Option of the Node.js http.Agent constructor is documented here: https://nodejs.org/api/dns.html#dnslookuphostname-options-callback
> - family <integer> | <string> The record family. Must be 4, 6, or 0. For backward compatibility reasons,'IPv4' and 'IPv6' are interpreted as 4 and 6 respectively. The value 0 indicates that either an IPv4 or IPv6 address is returned. If the value 0 is used with { all: true } (see below), either one of or both IPv4 and IPv6 addresses are returned, depending on the system's DNS resolver. Default: 0.

I would get an PR ready if this is acceptable.

---

Introduced by: https://github.com/module-federation/core/commit/f6381e630801a00543cd49ae92293d200b08e2be
Related to: https://github.com/module-federation/core/issues/3382

### Reproduction
https://github.com/module-federation/vite/tree/main/examples/vite-vite

host with: `dts: { generateTypes: false, consumeTypes: true }`
remote with `dts: { generateTypes: true, consumeTypes: false }`

### Used Package Manager

pnpm

### System Info
\-

### Validations

- [x] Read the [docs](https://github.com/module-federation/core).
- [x] Read the [common issues list](https://github.com/module-federation/core/issues).
- [x] Check that there isn't [already an issue](https://github.com/module-federation/core/issues) that reports the same bug to avoid creating a duplicate.
- [x] Make sure this is a Module federation issue and not a framework-specific issue.
- [x] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/.changeset/warm-dts-windows-paths.md
+++ b/.changeset/warm-dts-windows-paths.md
@@ -1,0 +1,6 @@
+---
+'@module-federation/dts-plugin': patch
+'@module-federation/sdk': patch
+---
+
+Fix Windows manifest type URL resolution in dts-plugin consumeTypes and allow the default IP family option to auto-select with `0`.

--- a/apps/website-new/docs/en/configure/dts.mdx
+++ b/apps/website-new/docs/en/configure/dts.mdx
@@ -223,7 +223,7 @@ interface DtsHostOptions {
   deleteTypesFolder?: boolean;
   maxRetries?: number;
   consumeAPITypes?: boolean;
-  family?: 4 | 6;
+  family?: 0 | 4 | 6;
 }
 ```
 
@@ -356,9 +356,9 @@ export default createModuleFederationConfig({
 
 #### family
 
-- Type: `4 | 6`
+- Type: `0 | 4 | 6`
 - Required: No
-- Default value: 4
+- Default value: 0
 
 Configure the IP version family that will be used for network operations.
 

--- a/apps/website-new/docs/zh/configure/dts.mdx
+++ b/apps/website-new/docs/zh/configure/dts.mdx
@@ -353,9 +353,9 @@ export default createModuleFederationConfig({
 ```
 #### family
 
-- Type: `4 | 6`
+- Type: `0 | 4 | 6`
 - Required: No
-- Default value: 4
+- Default value: 0
 
 配置 ip 版本，用于加载生产者类型文件。
 

--- a/packages/dts-plugin/src/core/configurations/hostPlugin.test.ts
+++ b/packages/dts-plugin/src/core/configurations/hostPlugin.test.ts
@@ -43,7 +43,7 @@ describe('hostPlugin', () => {
           runtimePkgs: [],
           remoteTypeUrls: {},
           timeout: 60000,
-          family: 4,
+          family: 0,
           typesOnBuild: false,
         });
 
@@ -102,6 +102,17 @@ describe('hostPlugin', () => {
             'moduleFederationTypescript',
           ),
         ).toStrictEqual(destinationPath);
+      });
+
+      it('preserves an explicit IPv6 family override', () => {
+        const options = {
+          moduleFederationConfig,
+          family: 6 as const,
+        };
+
+        const { hostOptions } = retrieveHostConfig(options);
+
+        expect(hostOptions.family).toBe(6);
       });
     });
 

--- a/packages/dts-plugin/src/core/configurations/hostPlugin.ts
+++ b/packages/dts-plugin/src/core/configurations/hostPlugin.ts
@@ -21,7 +21,7 @@ const defaultOptions = {
   remoteTypeUrls: {},
   timeout: 60000,
   typesOnBuild: false,
-  family: 4,
+  family: 0,
 } satisfies Partial<HostOptions>;
 
 const buildZipUrl = (hostOptions: Required<HostOptions>, url: string) => {

--- a/packages/dts-plugin/src/core/lib/DTSManager.general.spec.ts
+++ b/packages/dts-plugin/src/core/lib/DTSManager.general.spec.ts
@@ -160,6 +160,46 @@ describe('DTSManager General Tests', () => {
       expect(result.zipUrl).toContain('http://example.com/types.zip');
     });
 
+    it('should preserve manifest type URLs under Windows path semantics', async () => {
+      const manifestResponse = {
+        data: {
+          metaData: {
+            types: {
+              zip: '@mf-types.zip',
+              api: '@mf-types.d.ts',
+            },
+            publicPath: 'http://localhost:8090',
+          },
+        },
+      };
+
+      vi.spyOn(utils, 'nativeFetch').mockResolvedValueOnce({
+        data: manifestResponse.data,
+        status: 200,
+        headers: {},
+      } as any);
+
+      const remoteInfo: RemoteInfo = {
+        name: 'test',
+        url: 'http://localhost:8090/mf-manifest.json',
+        alias: 'test-alias',
+      };
+
+      const joinSpy = vi
+        .spyOn(path, 'join')
+        .mockImplementation(path.win32.join as typeof path.join);
+
+      try {
+        // @ts-expect-error only need timeout, which is not required
+        const result = await dtsManager.requestRemoteManifest(remoteInfo, {});
+
+        expect(result.zipUrl).toBe('http://localhost:8090/@mf-types.zip');
+        expect(result.apiTypeUrl).toBe('http://localhost:8090/@mf-types.d.ts');
+      } finally {
+        joinSpy.mockRestore();
+      }
+    });
+
     it('should handle manifest URLs without API types', async () => {
       const manifestResponse = {
         data: {

--- a/packages/dts-plugin/src/core/lib/DTSManager.ts
+++ b/packages/dts-plugin/src/core/lib/DTSManager.ts
@@ -255,8 +255,13 @@ class DTSManager {
         publicPath = inferAutoPublicPath(remoteInfo.url);
       }
 
+      const normalizedPublicPath = addProtocol(publicPath).endsWith('/')
+        ? addProtocol(publicPath)
+        : `${addProtocol(publicPath)}/`;
+
       remoteInfo.zipUrl = new URL(
-        path.join(addProtocol(publicPath), manifestJson.metaData.types.zip),
+        manifestJson.metaData.types.zip,
+        normalizedPublicPath,
       ).href;
       if (!manifestJson.metaData.types.api) {
         console.warn(`Can not get ${remoteInfo.name}'s api types url!`);
@@ -264,7 +269,8 @@ class DTSManager {
         return remoteInfo as Required<RemoteInfo>;
       }
       remoteInfo.apiTypeUrl = new URL(
-        path.join(addProtocol(publicPath), manifestJson.metaData.types.api),
+        manifestJson.metaData.types.api,
+        normalizedPublicPath,
       ).href;
       return remoteInfo as Required<RemoteInfo>;
     } catch (_err) {

--- a/packages/dts-plugin/src/core/lib/utils.ts
+++ b/packages/dts-plugin/src/core/lib/utils.ts
@@ -166,7 +166,7 @@ const getEnvHeaders = (): Record<string, string> => {
 
 export type FetchRequestConfig = {
   timeout?: number;
-  family?: 4 | 6;
+  family?: 0 | 4 | 6;
   headers?: Record<string, string>;
   responseType?: 'arraybuffer';
   /**
@@ -181,7 +181,7 @@ export type FetchRequestConfig = {
   agent?: unknown;
 };
 
-const createDispatcherFromFamily = (family?: 4 | 6): DispatcherLike => {
+const createDispatcherFromFamily = (family?: 0 | 4 | 6): DispatcherLike => {
   if (!family) return undefined;
   if (dispatcherCache.has(family)) return dispatcherCache.get(family);
   try {

--- a/packages/enhanced/src/schemas/container/ModuleFederationPlugin.check.ts
+++ b/packages/enhanced/src/schemas/container/ModuleFederationPlugin.check.ts
@@ -444,7 +444,7 @@ const t = {
                         ],
                       },
                       timeout: { type: 'number' },
-                      family: { enum: [4, 6] },
+                      family: { enum: [0, 4, 6] },
                       typesOnBuild: { type: 'boolean' },
                     },
                   },
@@ -4361,6 +4361,8 @@ function D(
                                                                             const n =
                                                                               c;
                                                                             if (
+                                                                              0 !==
+                                                                                e &&
                                                                               4 !==
                                                                                 e &&
                                                                               6 !==

--- a/packages/enhanced/src/schemas/container/ModuleFederationPlugin.ts
+++ b/packages/enhanced/src/schemas/container/ModuleFederationPlugin.ts
@@ -936,7 +936,7 @@ export default {
                       type: 'number',
                     },
                     family: {
-                      enum: [4, 6],
+                      enum: [0, 4, 6],
                     },
                     typesOnBuild: {
                       type: 'boolean',

--- a/packages/sdk/src/types/plugins/ModuleFederationPlugin.ts
+++ b/packages/sdk/src/types/plugins/ModuleFederationPlugin.ts
@@ -382,7 +382,7 @@ export interface DtsHostOptions {
   remoteTypeUrls?: (() => Promise<RemoteTypeUrls>) | RemoteTypeUrls;
   timeout?: number;
   /** The family of IP, used for network requests */
-  family?: 4 | 6;
+  family?: 0 | 4 | 6;
   typesOnBuild?: boolean;
 }
 

